src/main/java/com/google/sps/server/AddSchedule.java

```diff
@@ -0,0 +1,212 @@
+        List<TimeRange> sectionTimes = currentCourse.getSections().get(0).getMeetingTimes();
+        for (int i = 0; i < sectionTimes.size(); i++) {
+          String startTime = calculateStartTime(sectionTimes.get(i), month, year, nextSun);
+          String endTime = calculateEndTime(sectionTimes.get(i), month, year, nextSun);
+          addEvent(currentCourse, client, calendarId, startTime, endTime);
+        }
+      }       
+    } catch (IOException e) {
+      System.out.println("Couldn't iterate through the courses in the schedule.");
+    }
+  }
+
+   /**
+   * Helper function to calculate the day of the month of the next Sunday,
+   * to know when to start the schedule. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * 
+   * @return an int representing the date of next Sunday
+   */
+  public static int calculateDateOfNextSunday(java.util.Calendar calendar) {
+    int currentDay = calendar.get(java.util.Calendar.DAY_OF_MONTH);
+    int currentDayOfWeek = calendar.get(java.util.Calendar.DAY_OF_WEEK);
+    int nextSun = currentDay + 8 - currentDayOfWeek;
+    return nextSun;
+  }
+ 
+   /**
+   * Helper function to know the current month. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * 
+   * @return an int representing the month, adjusted for DateTime's formatting
+   */
+  public static int getCurrentMonth(java.util.Calendar calendar) {
+    return calendar.get(java.util.Calendar.MONTH) + 1;
+  }
+  
+   /**
+   * Helper function to know the current year. 
+   *
+   * @param calendar an instance of Calendar, so can know the current date
+   * 
+   * @return an int representing the current year
+   */
+  public static int getCurrentYear(java.util.Calendar calendar) {
+    return calendar.get(java.util.Calendar.YEAR);
+  }
+  
+   /**
+   * Helper function to format the relevant times (start or end) for a course 
+   * being converted into a calendar event. 
+   *
+   * @param timeInMin the time, in minutes of the week, for an event (either the start or end time)
+   * @param month the current month
+   * @param year the current year
+   * @param nextSun the day of the month of the next Sunday
+   * 
+   * @return a String formatted for DateTime, to be used in creating a calendar event 
+   */
+  public static String calculateDateTime(int timeInMin, int month, int year, int nextSun) {
+    int dayOfWeek = (int) Math.floorDiv(timeInMin, 24 * 60);
+    int dayInHours = dayOfWeek * 24;
+    int hour = (int) Math.floor((timeInMin / 60) - dayInHours);
+    int minute = (int) (60 * (((timeInMin / 60.0) - dayInHours) - hour));
+
+    int day = dayOfWeek + nextSun;
+    String dayString = "0" + Integer.toString(day);
+    if (day > 9) {
+      dayString = Integer.toString(day);
+    }
+
+    String monthString = "0" + Integer.toString(month);
+    if (month > 9) {
+      monthString = Integer.toString(month);
+    }
```

Comment:
Not a necessary change, but a cool little "pretty" thing that you could do to make these more compact. Java's Ternary operator would be useful here, something like 
`String monthString = month > 9 ? Integer.toString(month) : "0" + Integer.toString(month)`
or 
```
String monthString = month > 9 
  ? Integer.toString(month) 
  : "0" + Integer.toString(month)
```
(These are the same thing, just different formatting)

Reply:
Thanks for the suggestion, that's definitely nicer than what I have, I'll change that.